src/expr.cc

```diff
@@ -371,12 +372,31 @@ MemberLookup::MemberLookup(Expression *expr, const std::string &member, const Lo
 Value MemberLookup::evaluate(const std::shared_ptr<const Context>& context) const
 {
     const Value &v = this->expr->evaluate(context);
+    static const boost::regex re_swizzle_validation("^([xyzw]{1,4}|[rgba]{1,4})$");
 
     switch(v.type()) {
     case Value::Type::VECTOR:
+        if (this->member.length() > 1 && boost::regex_match(this->member, re_swizzle_validation))
+        {
+          VectorType ret(context->session());
+          for(const char& ch : this->member)
+            switch(ch)
+            {
+              case 'r': case 'x': ret.emplace_back(v[0]); break;
+              case 'g': case 'y': ret.emplace_back(v[1]); break;
+              case 'b': case 'z': ret.emplace_back(v[2]); break;
+              case 'a': case 'w': ret.emplace_back(v[3]); break;
+            }
+          return ret;
```

Comment:
This triggers a problem with the Ubuntu 18.04. I assume that's because the older boost version does not support the automatic type conversion. This should fix the build issue:
```suggestion
          return Value(std::move(ret));
```